graphql_compiler/scripts/generate_test_sql/animals.py

```diff
@@ -36,6 +36,16 @@ def _get_initial_animal_generators(species_name, current_animal_names):
     return generator_list
 
 
+def _get_new_parents(current_animal_names, previous_parent_sets):
+    """Return a list of `NUM_PARENTS` parent names that is not present in `previous_parent_sets`."""
```

Comment:
let's make `NUM_PARENTS` an input variable, no need to introduce global state into this function